BrainPortal/app/models/boutiques_cluster_task.rb

```diff
@@ -83,15 +83,17 @@ def setup #:nodoc:
 
       # Most common situation
       if ! input.list || ! userfile.is_a?(CbrainFileList)
-        make_available(userfile, userfile.name)
+        copy_file = descriptor.custom.dig('cbrain:full-copy-input-files')&.include?(input.id)
```

Comment:
You are extracting this option twice, once at line 86 and once at line 95, and at line 95 you are also doing it within an internal loop. This flag doesn't change over time, extract it just once for the input, at around line 83!